### PR TITLE
feature : add EDITOR_NOT_IN_PREVIEW marco

### PR DESCRIPTION
**Pull request by dogeFu**
Re: 
- https://github.com/cocos/3d-tasks/issues/16811
 
Depend:
- https://github.com/cocos/cocos-editor/pull/2123
### Changelog
* add EDITOR_NOT_IN_PREVIEW marco
* replace GAME_VIEW condition with ERITOR_NOT_IN_PREVIEW marco 
   * EDITOR && !ccleagcy.GAME_VIEW => ERITOR_NOT_IN_PREVIEW
   * !EDITOR || ccleagcy.GAME_VIEW => !EDITOR_NOT_IN_PREVIEW
* remove unuse import
-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [x] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/2d/utils/dynamic-atlas/atlas-manager.ts [hunk 1]
```diff
@@ -273,7 +273,7 @@ export class DynamicAtlasManager extends System {
      * @param frame  the sprite frame that will be packed in the dynamic atlas.
      */
     public packToDynamicAtlas (comp, frame) {
-        if ((EDITOR && !cclegacy.GAME_VIEW) || !this._enabled) return;
+        if ((EDITOR_NOT_IN_PREVIEW) || !this._enabled) return;
```
Comment (jareguo) [inline cocos/2d/utils/dynamic-atlas/atlas-manager.ts]: There's an extra parenthesis

File: cocos/animation/animation-component.ts [hunk 2]
```diff
@@ -170,7 +170,7 @@ export class Animation extends Eventify(Component) {
     }
 
     public start () {
-        if ((!EDITOR || cclegacy.GAME_VIEW) && (this.playOnLoad && !this._hasBeenPlayed) && this._defaultClip) {
+        if ((!EDITOR_NOT_IN_PREVIEW) && (this.playOnLoad && !this._hasBeenPlayed) && this._defaultClip) {
```
Comment (jareguo) [inline cocos/animation/animation-component.ts]: There's an extra parenthesis

File: cocos/asset/asset-manager/config.ts [hunk 3]
```diff
@@ -148,7 +147,7 @@ const isMatchByWord = (path: string, test: string): boolean => {
 };
 
 const processOptions = (options: IConfigOption) => {
-    if ((EDITOR && !legacyCC.GAME_VIEW) || TEST) { return; }
+    if ((EDITOR_NOT_IN_PREVIEW) || TEST) { return; }
```
Comment (jareguo) [inline cocos/asset/asset-manager/config.ts]: There's an extra parenthesis

**Review by jareguo — APPROVED** [review]

File: cc.config.json [hunk 4]
```diff
@@ -546,6 +546,14 @@
             "internal": false,
             "dynamic": true
         },
+        "EDITOR_NOT_IN_PREVIEW" : {
+            "comment": "Run in editor but not in editor preview.",
+            "type": "boolean",
+            "value": false,
+            "ccGlobal": true,
```
Comment (SantyWang) [inline cc.config.json]: ccGlobal is for 2.x compliant implementations and should be set to false here.

Comment (dogeFu) [reply cc.config.json]: done

**Review by PPpro — APPROVED** [review]

**Review by shrinktofit — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

**Pull request merged**